### PR TITLE
Dialog renderer implementation for subscriptions

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: extensions/amp-subscriptions/0.1/amp-subscriptions.css [hunk 1]
```diff
@@ -75,3 +75,51 @@ html:not(.i-amphtml-subs-loggedin-yes) [subscriptions-action="logout"] {
 html:not(.i-amphtml-subs-subscriber-no) [subscriptions-action="subscribe"] {
   display: none !important;
 }
+
+
+/** Dialog. */
+amp-subscriptions-dialog {
+  display: block !important;
```
Comment (prateekbh) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.css]: isn't `width:100%` and `display:block` redundant? as `position:fixed` already makes element behave `display:inline`.

Or is this supposed to be `display:none` somewhere which this line overrides?

File: extensions/amp-subscriptions/0.1/amp-subscriptions.css [hunk 2]
```diff
@@ -75,3 +75,51 @@ html:not(.i-amphtml-subs-loggedin-yes) [subscriptions-action="logout"] {
...
+  position: fixed !important;
+  bottom: 0 !important;
+  left: 0;
```
Comment (prateekbh) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.css]: no `!important`?

File: extensions/amp-subscriptions/0.1/dialog-renderer.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  render(authResponse) {
+    // Make sure the document is fully parsed.
+    return this.ampdoc_.whenReady().then(() => {
+      // Find the first matching dialog.
+      const candidates = this.ampdoc_.getRootNode()
+          .querySelectorAll('[subscriptions-dialog][subscriptions-display]');
+      for (let i = 0; i < candidates.length; i++) {
+        const candidate = candidates[i];
+        const expr = candidate.getAttribute('subscriptions-display');
+        if (expr && evaluateExpr(expr, authResponse)) {
```
Comment (prateekbh) [inline extensions/amp-subscriptions/0.1/dialog-renderer.js]: Do we expect `authResponse` to be something like `{value: X}`? if Yes is this the best name for this variable?
Do we have a defined structure of authResponse?

File: extensions/amp-subscriptions/0.1/amp-subscriptions.css (hunk 1 above)
Comment (dvoytenko) [reply extensions/amp-subscriptions/0.1/amp-subscriptions.css]: No, `display:block` is still necessary, even if `position:fixed` defined. There are definitely nuances w/o specifying it.

File: extensions/amp-subscriptions/0.1/dialog-renderer.js (hunk 3 above)
Comment (dvoytenko) [reply extensions/amp-subscriptions/0.1/dialog-renderer.js]: Yes. It's essentially the `Entitlement.json()` call. I just didn't get there yet. The expressions themselves do not care - they simply built around json.

File: extensions/amp-subscriptions/0.1/amp-subscriptions.css (hunk 2 above)
Comment (dvoytenko) [reply extensions/amp-subscriptions/0.1/amp-subscriptions.css]: Fixed

**Pull request merged**